vllm/model_executor/custom_op.py

```diff
@@ -1,52 +1,106 @@
+import torch
 import torch.nn as nn
 
+import vllm.envs as envs
 from vllm.utils import is_cpu, is_hip, is_tpu, is_xpu
 
 
 class CustomOp(nn.Module):
 
     def __init__(self, *args, **kwargs):
         super().__init__()
+        self._is_compiled = False
         self._forward_method = self.dispatch_forward()
 
     def forward(self, *args, **kwargs):
         return self._forward_method(*args, **kwargs)
 
-    def forward_native(self, *args, **kwargs):
+    @staticmethod
+    def forward_static(*args, **kwargs):
         """PyTorch-native implementation of the forward method.
 
-        This method is optional. If implemented, it can be used with compilers
-        such as torch.compile or PyTorch XLA. Also, it can be used for testing
-        purposes.
+        This method is optional. If implemented, it can be used with
+        `torch.compile` to generate the custom kernel for the op.
+        This function is necessary because any method that uses `self`
+        triggers re-compilation in `torch.compile`.
```

Comment:
This should go away with PyTorch 2.5 (its ok to still go through this PR, it if unblocks you). For 2.4, you can probably to `torch._dynamo.config.inline_inbuilt_nn_modules = True` and that will prevent recompilations on `self`.

Reply:
@anijain2305 Thanks for the info! Update the PR.